File: 8kyu/(8 kyu) String cleaning/(8 kyu) String cleaning.js
// 1 Plain solution
function stringClean(s) {
    let res = '';

    for (let i = 0; i < s.length; i++) {
        if (s.charCodeAt(i) < 48 || s.charCodeAt(i) > 57) {
            res += s[i];
        }
    }

    return res;
}

// 2 Optimized solution
function stringClean(s) {
    return [...s]
        .filter(
            c => c.charCodeAt(0) < 48
                || c.charCodeAt(0) > 57
        )
        .join('');
}

// 3 Clever solution
const stringClean = s => s.replace(/\d/g, "");

File: 8kyu/(8 kyu) String cleaning/(8 kyu) String cleaning.py
# 1 Plain solution
def string_clean(s):
    res = ''

    for char in s:
        if not char.isdigit():
            res += char

    return res

# 2 Optimized solution
def string_clean(s):
    return ''.join(filter(lambda x: not x.isdigit(), s))

# 3 Clever solution
import re
def string_clean(s):
    return re.sub(r'\d', '', s)